lib/emerald/parser.rb

```diff
@@ -27,9 +37,15 @@ def parse_node(source)
       when /[\d]/
         parse_number(source)
       when /[+-]/
```

Comment:
Would be nice to restructure it in some way like that though, rather than the nested if.